src/components/CustomCollection/CustomCollection.tsx

```diff
@@ -387,8 +387,9 @@ const CustomCollection = ({
             <Collection
               header={customCollectionHeader}
               footer={!isEditingTitle ? addLinkForm : null}>
-              {visibleBookmarks.map((bookmark: BookmarkType, index) =>
-                bookmark.cmsId ? (
+              {visibleBookmarks.map((bookmark: MongoBookmark, index) => {
+                const foundBookmark = findBookmark(bookmark)
+                return bookmark.cmsId && foundBookmark ? (
```

Comment:
I opted to add this check for an undefined bookmark here so that we could maintain the new and improved type-safe `RemovableBookmark` component. So now if there is some misalignment between `cmsId` and `id` it will just display as a `CustomBookmark`